src/common/iop_order.h

```diff
@@ -158,83 +158,83 @@ typedef struct dt_iop_order_rule_t
+GList *dt_ioppr_merge_multi_instance_iop_order_list(GList *const __restrict__ iop_order_list, GList *const __restrict__ multi_instance_list);
 
 /** returns 1 if there's a module_so without a iop_order defined */
-int dt_ioppr_check_so_iop_order(GList *iop_list, GList *iop_order_list);
+int dt_ioppr_check_so_iop_order(GList *const __restrict__ iop_list, GList *const __restrict__ iop_order_list);
 
 /* returns a list of dt_iop_order_rule_t with the current iop order rules */
 GList *dt_ioppr_get_iop_order_rules();
 
 /** returns a duplicate of iop_order_list */
-GList *dt_ioppr_iop_order_copy_deep(GList *iop_order_list);
+GList *dt_ioppr_iop_order_copy_deep(GList *const iop_order_list);
 
 /** sort two modules by iop_order */
 gint dt_sort_iop_by_order(gconstpointer a, gconstpointer b);
 gint dt_sort_iop_list_by_order_f(gconstpointer a, gconstpointer b);
 
 /** returns the iop_order before module_next if module can be moved */
-gboolean dt_ioppr_check_can_move_before_iop(GList *iop_list, struct dt_iop_module_t *module, struct dt_iop_module_t *module_next);
+gboolean dt_ioppr_check_can_move_before_iop(GList *const iop_list, struct dt_iop_module_t *const __restrict__module, struct dt_iop_module_t *const __restrict__ module_next);
 /** returns the iop_order after module_prev if module can be moved */
-gboolean dt_ioppr_check_can_move_after_iop(GList *iop_list, struct dt_iop_module_t *module, struct dt_iop_module_t *module_prev);
+gboolean dt_ioppr_check_can_move_after_iop(GList *const iop_list, struct dt_iop_module_t *const __restrict__ module, struct dt_iop_module_t *const __restrict__ module_prev);
 
 /** moves module before/after module_next/previous on pipe */
-gboolean dt_ioppr_move_iop_before(struct dt_develop_t *dev, struct dt_iop_module_t *module, struct dt_iop_module_t *module_next);
-gboolean dt_ioppr_move_iop_after(struct dt_develop_t *dev, struct dt_iop_module_t *module, struct dt_iop_module_t *module_prev);
+gboolean dt_ioppr_move_iop_before(struct dt_develop_t *const dev, struct dt_iop_module_t *const __restrict__ module, struct dt_iop_module_t *const __restrict__ module_next);
+gboolean dt_ioppr_move_iop_after(struct dt_develop_t *const dev, struct dt_iop_module_t *const __restrict__ module, struct dt_iop_module_t *const __restrict__ module_prev);
 
 // for debug only
-int dt_ioppr_check_iop_order(struct dt_develop_t *dev, const int imgid, const char *msg);
-void dt_ioppr_print_module_iop_order(GList *iop_list, const char *msg);
-void dt_ioppr_print_history_iop_order(GList *history_list, const char *msg);
-void dt_ioppr_print_iop_order(GList *iop_order_list, const char *msg);
+int dt_ioppr_check_iop_order(struct dt_develop_t *const dev, const int imgid, const char *msg);
+void dt_ioppr_print_module_iop_order(GList *const iop_list, const char *msg);
+void dt_ioppr_print_history_iop_order(GList *const history_list, const char *msg);
+void dt_ioppr_print_iop_order(GList *const iop_order_list, const char *msg);
```

Comment:
Const in function _declarations_ is redundant and can be omitted (const to declare the variable or pointer itself as const, not to indicate a pointer that points to something constant). This means, it is absolute valid to declare 

`int dt_ioppr_check_iop_order(struct dt_develop_t *dev, int imgid, const char *msg);`

and to implement

`int dt_ioppr_check_iop_order(struct dt_develop_t *const dev, int imgid, const char *msg) { ... };`

Reply:
Ok, thanks for the tips. It's really stupid that `f(const *x)` needs to go in the header, but `f(*const x)` is as good as nothing. Same with `restrict`. Such an inconsistent language… What about `__attribute__((aligned(64)))` ?

Reply:
I think you didn't get the point. *const and restrict does nothing in the header, because it matters only for implementation, so you can use them in definition. In declaration they just don't mean anything and are just noise.
Not sure what __attribute__((aligned(64))) has to do with anything. It makes no sense in function prototype.